src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1954,37 +1986,50 @@ private void VisitArgumentEvaluate(ImmutableArray<BoundExpression> arguments, Im
             }
         }
 
-        private static ImmutableArray<BoundExpression> RemoveArgumentConversions(ImmutableArray<BoundExpression> arguments, ImmutableArray<RefKind> refKindsOpt)
+        private static (ImmutableArray<BoundExpression> Arguments, ImmutableArray<Conversion> Conversions) RemoveArgumentConversions(
+            ImmutableArray<BoundExpression> arguments,
+            ImmutableArray<RefKind> refKindsOpt)
         {
             int n = arguments.Length;
+            var conversions = default(ImmutableArray<Conversion>);
             if (n > 0)
             {
                 var argumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(n);
+                var conversionsBuilder = ArrayBuilder<Conversion>.GetInstance(n);
                 bool includedConversion = false;
                 for (int i = 0; i < n; i++)
                 {
                     RefKind refKind = GetRefKind(refKindsOpt, i);
                     var argument = arguments[i];
+                    var conversion = Conversion.Identity;
+                    // PROTOTYPE(NullableReferenceTypes): Should `RefKind.In` be treated similarly to `RefKind.None`?
                     if (refKind == RefKind.None)
                     {
-                        argument = RemoveImplicitConversions(argument);
+                        (argument, conversion) = RemoveConversion(argument, includeExplicitConversions: false);
                         if (argument != arguments[i])
```

Comment:
>arguments[i] [](start = 40, length = 12)

[Nit] The only reason you need to index the array again is that you've overwritten the variable `argument` that you previously read the element into. Perhaps better to read into a fresh variable so you can compare two locale.